src/audio/copier/copier.c

```diff
@@ -561,6 +561,8 @@ static int copier_multi_endpoint_dai_copy(struct copier_data *cd, struct comp_de
 
 	ret = dai_zephyr_multi_endpoint_copy(cd->dd, dev, cd->multi_endpoint_buffer,
 					     cd->endpoint_num);
+	if (!ret)
+		comp_update_buffer_consume(src_c, processed_data.source_bytes);
```

Comment:
does `.input_total_data_processed` need to be updated too?

Reply:
I need to say:  "I think so". according to origin code logic, we need update this as well without extra conditions.
since it is multi dai case, endpoint_num != 0 and bsource_buffer == true.

These part of code is still in critical path, we'd better to find a way to remove some checks.